File: app/src/main/java/com/lifedawn/bestweather/ui/notification/daily/DailyPushNotificationReceiver.java
package com.lifedawn.bestweather.ui.notification.daily;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.work.Data;
import androidx.work.ExistingWorkPolicy;
import androidx.work.OneTimeWorkRequest;
import androidx.work.OutOfQuotaPolicy;
import androidx.work.WorkManager;

import com.lifedawn.bestweather.commons.constants.BundleKey;
import com.lifedawn.bestweather.ui.notification.daily.work.DailyNotificationListenableWorker;

public class DailyPushNotificationReceiver extends BroadcastReceiver {


	@Override
	public void onReceive(Context context, Intent intent) {
		final String action = intent.getAction();

		if (action != null) {
			Bundle arguments = intent.getExtras();
			final int id = arguments.getInt(BundleKey.dtoId.name());

			Data data = new Data.Builder()
					.putString("DailyPushNotificationType", arguments.getString(
							"DailyPushNotificationType"))
					.putInt(BundleKey.dtoId.name(), id)
					.putString("action", action)
					.build();

			final String tag = DailyNotificationListenableWorker.class.getName() + action;

			OneTimeWorkRequest request = new OneTimeWorkRequest.Builder(DailyNotificationListenableWorker.class)
					.setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
					.setInputData(data)
					.addTag(tag)
					.build();

			WorkManager workManager = WorkManager.getInstance(context);
			workManager.enqueueUniqueWork(tag, ExistingWorkPolicy.REPLACE, request);
		}
	}


}

File: app/src/main/java/com/lifedawn/bestweather/ui/notification/daily/DailyPushNotificationReceiver.kt
package com.lifedawn.bestweather.ui.notification.daily

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.work.*
import com.lifedawn.bestweather.commons.constants.BundleKey
import com.lifedawn.bestweather.ui.notification.daily.work.DailyNotificationListenableWorker

class DailyPushNotificationReceiver constructor() : BroadcastReceiver() {
    public override fun onReceive(context: Context, intent: Intent) {
        val action: String? = intent.getAction()
        if (action != null) {
            val arguments: Bundle? = intent.getExtras()
            val id: Int = arguments!!.getInt(BundleKey.dtoId.name)
            val data: Data = Data.Builder()
                .putString(
                    "DailyPushNotificationType", arguments.getString(
                        "DailyPushNotificationType"
                    )
                )
                .putInt(BundleKey.dtoId.name, id)
                .putString("action", action)
                .build()
            val tag: String = DailyNotificationListenableWorker::class.java.getName() + action
            val request: OneTimeWorkRequest = OneTimeWorkRequest.Builder(DailyNotificationListenableWorker::class.java)
                .setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
                .setInputData(data)
                .addTag(tag)
                .build()
            val workManager: WorkManager = WorkManager.getInstance(context)
            workManager.enqueueUniqueWork(tag, ExistingWorkPolicy.REPLACE, request)
        }
    }
}